sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteTables.java

```diff
@@ -198,22 +189,11 @@ private void load(
             BigQueryHelpers.jobToPrettyString(lastFailedLoadJob)));
   }
 
-  static void removeTemporaryFiles(
-      PipelineOptions options, String tempFilePrefix, Collection<String> files) throws IOException {
-    IOChannelFactory factory = IOChannelUtils.getFactory(tempFilePrefix);
-    if (factory instanceof GcsIOChannelFactory) {
-      GcsUtil gcsUtil = new GcsUtilFactory().create(options);
-      gcsUtil.remove(files);
-    } else if (factory instanceof FileIOChannelFactory) {
-      for (String filename : files) {
-        LOG.debug("Removing file {}", filename);
-        boolean exists = Files.deleteIfExists(Paths.get(filename));
-        if (!exists) {
-          LOG.debug("{} does not exist.", filename);
-        }
-      }
-    } else {
-      throw new IOException("Unrecognized file system.");
+  static void removeTemporaryFiles(Collection<String> files) throws IOException {
```

Comment:
That would move the conversion from `String` to `ResourceId` at two places which produce filenames as `String`s